assets/js/modules/analytics-4/datastore/audience-settings.js

```diff
@@ -0,0 +1,323 @@
+			}
+
+			return { response, error };
+		}
+	),
+
+	/**
+	 * Sets the configured audiences.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Array} audienceResourceNames Configured audience resource names.
+	 * @return {Object} Redux-style action.
+	 */
+	setConfiguredAudiences( audienceResourceNames ) {
+		invariant(
+			Array.isArray( audienceResourceNames ),
+			'Configured audiences should be an array.'
+		);
+
+		return {
+			type: SET_CONFIGURED_AUDIENCES,
+			payload: { audienceResourceNames },
+		};
+	},
+
+	/**
+	 * Sets the audience segmentation widget visibility.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {boolean} isWidgetHidden Whether or not the audience segmentation widget is hidden.
+	 * @return {Object} Redux-style action.
+	 */
+	setAudienceSegmentationWidgetHidden( isWidgetHidden ) {
+		invariant(
+			typeof isWidgetHidden === 'boolean',
+			'Audience segmentation widget visibility should be a boolean.'
+		);
+
+		return {
+			type: SET_AUDIENCE_SEGMENTATION_WIDGET_HIDDEN,
+			payload: { isWidgetHidden },
+		};
+	},
+};
+
+const baseControls = {};
+
+const baseReducer = createReducer( ( state, { type, payload } ) => {
+	switch ( type ) {
+		case SET_CONFIGURED_AUDIENCES: {
+			const { audienceResourceNames } = payload;
+
+			if ( ! state.audienceSettings ) {
+				state.audienceSettings = {};
+			}
+
+			state.audienceSettings.settings = {
+				...state.audienceSettings.settings,
+				configuredAudiences: audienceResourceNames,
+			};
+
+			break;
+		}
+
+		case SET_AUDIENCE_SEGMENTATION_WIDGET_HIDDEN: {
+			const { isWidgetHidden } = payload;
+
+			if ( ! state.audienceSettings ) {
+				state.audienceSettings = {};
+			}
+
+			state.audienceSettings.settings = {
+				...state.audienceSettings.settings,
+				isAudienceSegmentationWidgetHidden: isWidgetHidden,
+			};
+
+			break;
+		}
+
+		default: {
+			return state;
+		}
+	}
+} );
+
+const baseResolvers = {
+	*getAudienceSettings() {
+		const registry = yield Data.commonActions.getRegistry();
+
+		const audienceSettings = registry
+			.select( MODULES_ANALYTICS_4 )
+			.getAudienceSettings();
+
+		if ( audienceSettings === undefined ) {
+			yield fetchGetAudienceSettingsStore.actions.fetchGetAudienceSettings();
+		}
+	},
+};
+
+const baseSelectors = {
+	/**
+	 * Gets the audience settings.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {(Object|undefined)} Audience settings; `undefined` if not loaded.
+	 */
+	getAudienceSettings( state ) {
+		return state.audienceSettings?.settings;
+	},
+
+	/**
+	 * Gets the configured audiences from the audience settings.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {(Array|undefined)} An array with configured audiences; `undefined` if not loaded.
+	 */
+	getConfiguredAudiences: createRegistrySelector( ( select ) => () => {
+		const audienceSettings =
+			select( MODULES_ANALYTICS_4 ).getAudienceSettings();
```

Comment:
It would be more straightforward accessing the state directly `return state.audienceSettings?.settings?.configuredAudiences` instead of relying on another selector within a selector. WDYT?

Reply:
I thought so. However, we follow the convention of utilizing selectors even within the same partial store. More importantly, if the state is' undefined, ' it will invoke the resolver to fetch API.

Reply:
Makes sense, good to know so I can utilize the same in the future when needed. Thanks!